src/features/app-interfaces/data/write.ts

```diff
@@ -77,14 +61,14 @@ const createMachine = () =>
     types: {
       context: {} as {
         file?: File
-        appSpec?: Arc32AppSpec
+        appSpec?: Arc32AppSpec | Arc4AppSpec
```

Comment:
I think if you change the context of the state machine to
```
theRest & 
{
{ appSpecStandard: Arc-32, appSpec: Arc32AppSpec } | { appSpecStandard: Arc-4, appSpec: Arc4AppSpec }
}
```
You will be able to save many `isArc...` checks in the forms.